packages/steiger/src/shared/globs/apply-exclusion.spec.ts

```diff
@@ -0,0 +1,218 @@
+        📂 user
+          📄 index.ts
+        📂 post
+          📄 index.ts
+        `,
+      joinFromRoot('src'),
+    )
+
+    const globs = [
+      {
+        files: ['/src/shared/**', '/src/entities/**'],
+        ignores: ['**/ui/**'],
+      },
+    ]
+
+    expect(applyExclusion(vfs, globs)).toEqual(expectedVfs)
+  })
+
+  it('should correctly apply exclusions with an inverted glob group', () => {
+    const vfs = parseIntoFsdRoot(
+      `
+      📂 shared
+        📂 ui
+          📄 index.ts
+          📄 Button.tsx
+        📂 lib
+          📄 device-detect.ts
+          📄 get-query-params.ts
+          📄 index.ts
+      📂 entities
+        📂 user
+          📂 ui
+            📄 UserAvatar.tsx
+            📄 index.ts
+          📄 index.ts
+        📂 post
+          📂 ui
+            📄 index.ts
+            📄 PostList.tsx
+          📄 index.ts
+      `,
+      joinFromRoot('src'),
+    )
+
+    const expectedVfs = parseIntoFsdRoot(
+      `
+      📂 shared
+        📂 lib
+          📄 device-detect.ts
+          📄 index.ts`,
+      joinFromRoot('src'),
+    )
+
+    const globs = [
+      {
+        files: ['/src/shared/**', '/src/entities/**'],
+        ignores: ['**/ui/**'],
+      },
+      not({
+        files: ['/src/entities/**', '/src/shared/lib/get-query-params.ts'],
+        ignores: [],
+      }),
+    ]
+
+    expect(applyExclusion(vfs, globs)).toEqual(expectedVfs)
+  })
+
+  it('should correctly apply exclusions with several normal and inverted glob groups', () => {
+    const vfs = parseIntoFsdRoot(
+      `
+      📂 shared
+        📂 ui
+          📄 index.ts
+          📄 Button.tsx
+        📂 lib
+          📄 device-detect.ts
+          📄 get-query-params.ts
+          📄 index.ts
+      📂 entities
+        📂 user
+          📂 ui
+            📄 UserAvatar.tsx
+            📄 index.ts
+          📄 index.ts
+        📂 post
+          📂 ui
+            📄 index.ts
+            📄 PostList.tsx
+          📄 index.ts
+      📂 pages
+        📂 main
+          📄 index.ts
+        📂 account
+          📄 index.ts
+      `,
+      joinFromRoot('src'),
+    )
+
+    const expectedVfs = parseIntoFsdRoot(
+      `
+      📂 shared
+        📂 lib
+          📄 device-detect.ts
+          📄 index.ts
+      📂 pages
+        📂 main
+          📄 index.ts`,
+      joinFromRoot('src'),
+    )
+
+    const globs = [
+      {
+        files: ['/src/shared/**', '/src/entities/**'],
+        ignores: ['**/ui/**'],
+      },
+      not({
+        files: ['/src/entities/**', '/src/shared/lib/get-query-params.ts'],
+        ignores: [],
+      }),
+      {
+        files: ['/src/pages/**'],
+        ignores: [],
+      },
+      not({
+        files: ['/src/pages/account/**'],
+        ignores: [],
+      }),
+    ]
+
+    expect(applyExclusion(vfs, globs)).toEqual(expectedVfs)
+  })
+
+  it('should correctly apply exclusions for a special case', () => {
+    const globs = [not({ files: [], ignores: [] }), { files: ['/src/shared/ui/Button.ts'], ignores: [] }]
```

Comment:
question: what makes this case special?

Reply:
Yes, it looks like I should have renamed this test case, but I forgot to do so. Thanks for catching that!